### ISSUE TITLE
cannot move work item to doing / done columns

### ISSUE THREAD

**Issue by JamborJan** [edtro/EdTro.AzureDevOps.Extensions.querybasedboards#41]
Not sure if I got this wrong. I have this json config for my board as shown at the bottom. My question: why am I not able to drag n drop in one of the doing / done columns? In the picture below I try to drop teh item in teh analysis doing column and this didn't work.

<img width="781" alt="Bildschirmfoto 2021-02-19 um 16 58 11" src="https://user-images.githubusercontent.com/7508960/108530790-461ff780-72d6-11eb-85af-75833557178a.png">

```
{
   "settings":{
      "showFilter":true,
      "showArrows":true,
      "showProject":true,
      "showStoryPoints":false,
      "showTags":true
   },
   
   "swimlanes":{
      "field":"System.AreaPath",
      "defaultValue":"Planning",
      "values":[
         {
            "value":"Planning",
            "title":"Planning"
         },
         {
            "value":"Planning\\Platform",
            "title":"Platform"
         },
         {
            "value":"Planning\\Produkt1",
            "title":"Produkt1"
         },
         {
            "value":"Planning\\Produkt2",
            "title":"Produkt2"
         },
         {
            "value":"Planning\\Produkt3",
            "title":"Produkt3"
         }
      ]
   },
      "columns":[
      {
         "name":"Backlog",
         "title":"Backlog",
         "isBacklog":true
      },
      {
         "name":"Analysis",
         "title":"Analysis",
         "doingDone":{
            "field":"System.BoardColumnDone",
            "doingTitle":"Doing",
            "doingDefaultValue":false,
            "doneTitle":"Ready",
            "doneValue":true
         }
      },
      {
         "name":"Implementation",
         "title":"Implementation",
         "doingDone":{
            "field":"System.BoardColumnDone",
            "doingTitle":"Doing",
            "doingDefaultValue":false,
            "doneTitle":"Ready",
            "doneValue":true
         }
      },
      {
         "name":"Verification",
         "title":"Verification",
         "doingDone":{
            "field":"System.BoardColumnDone",
            "doingTitle":"Doing",
            "doingDefaultValue":false,
            "doneTitle":"Ready",
            "doneValue":true
         }
      },
      {
         "name":"Done",
         "title":"Done",
         "isBacklog":true
      }
   ]
}

```

I have a similar issue with task boards, json example and screen below. I cannot move the work item plus teh done (I call it ready) column ist not shown.

<img width="1189" alt="Bildschirmfoto 2021-02-19 um 17 11 46" src="https://user-images.githubusercontent.com/7508960/108530406-db6ebc00-72d5-11eb-95da-002e4209cf0e.png">

```
{
   "settings":{
      "showFilter":true,
      "showArrows":true,
      "showProject":true,
      "showStoryPoints":false,
      "showTags":true
   },
   
   "columns":[
      {
         "name":"Backlog",
         "title":"Backlog",
         "isBacklog":true
      },
      {
         "name":"Analysis",
         "title":"Analysis",
         "doingDone":{
            "field":"System.BoardColumnDone",
            "doingTitle":"Doing",
            "doingDefaultValue":false,
            "doneTitle":"Ready",
            "doneValue":true
         }
      },
      {
         "name":"Implementation",
         "title":"Implementation",
         "doingDone":{
            "field":"System.BoardColumnDone",
            "doingTitle":"Doing",
            "doingDefaultValue":false,
            "doneTitle":"Ready",
            "doneValue":true
         }
      },
      {
         "name":"Verification",
         "title":"Verification",
         "doingDone":{
            "field":"System.BoardColumnDone",
            "doingTitle":"Doing",
            "doingDefaultValue":false,
            "doneTitle":"Ready",
            "doneValue":true
         }
      },
      {
         "name":"Done",
         "title":"Done",
         "isBacklog":true
      }
   ]
}

```

**Comment by edtro** [comment]
Dear @JamborJan 
For the 'System.BoardColumnDone' the reason is simple. This is a system field and we are not allowed to modify this field while using the api or directly (I am investigating a workaround at the moment).

But all your questions can  be answered by this quote (from the documentation file config.md):

> NOTE: for obvious reasons all system fields are not allowed to be used e.g. System.WorkItemType, because the risk is too high or they are system protected.

So the System.IterationPath is not implemented.

Kr
Edward

**Comment by JamborJan** [comment]
I apologise for this rtfm-question. Sorry.

**Comment by JamborJan** [comment]
dear @edtro, I fixed most of this by implementing a custom boolean value to use the doing / done parts of the column. But on task boards I still have the issue, that the done part of the collumn is not shown. When I switch the boolean to done=true the work item simply disappears on the board. The strange thing is, the same column settings work on a single type board where I only have epics features etc. Only on task boards this one is not working.

Not working on epic-feature task board but working on feature only board:
```
{
   "settings":{
      "showFilter":true,
      "showArrows":true,
      "showProject":true,
      "showStoryPoints":false,
      "showTags":true
   },
   
   "columns":[
      {
         "name":"Backlog",
         "title":"Backlog",
         "isBacklog":true
      },
      {
         "name":"Analysis",
         "title":"Analysis",
         "doingDone":{
            "field":"Custom.IsReady",
            "doingTitle":"Doing",
            "doingDefaultValue":false,
            "doneTitle":"Ready",
            "doneValue":true
         }
      },
      {
         "name":"Implementation",
         "title":"Implementation",
         "doingDone":{
            "field":"Custom.IsReady",
            "doingTitle":"Doing",
            "doingDefaultValue":false,
            "doneTitle":"Ready",
            "doneValue":true
         }
      },
      {
         "name":"Verification",
         "title":"Verification",
         "doingDone":{
            "field":"Custom.IsReady",
            "doingTitle":"Doing",
            "doingDefaultValue":false,
            "doneTitle":"Ready",
            "doneValue":true
         }
      },
      {
         "name":"Done",
         "title":"Done",
         "isBacklog":true
      }
   ]
}

```

PS: I don't really get your comment "So the System.AreaPath is not implemented." because using this value for swimlanes seems to work pretty okay for me. Onyl a "Under" feature would be cool, so to have all area paths under "Planning\\Product\\*" in the swim lane. E.g. "Planning\\Product\\Team1" and "Planning\\Product\\Team2".

**Comment by edtro** [comment]
Ok, I will look into this (regarding the taskboard).

**Comment by edtro** [comment]
@JamborJan Hi, I think I have found & resolved the problem (regarding the doing/done within the taskboard, that it wouldnt show up). Now I have to test it a little further. Next to this... I have a lot of new stuff ready for release. But if everything goes well, as I hope it will, I will be able to release this soon.
Kr
Edward

**Comment by edtro** [comment]
Solved in 0.21059
